center/src/main/java/com/microsoft/hydralab/center/service/TestTaskService.java

```diff
@@ -82,25 +85,27 @@ public void runTask() {
             return;
         }
         isRunning.set(true);
-        while (!taskQueue.isEmpty()) {
-            TestTaskSpec testTaskSpec = taskQueue.peek();
+        for (TestTaskSpec testTaskSpec : taskQueue) {
             TestTask testTask = TestTask.convertToTestTask(testTaskSpec);
             try {
+                logger.info("Start triggering queued test task: " + testTaskSpec.testTaskId + ", target deviceIdentifier: " + testTaskSpec.deviceIdentifier);
```

Comment:
nit: "Start triggering queued test task" -> "Start trying to trigger queued test task". This may avoid confusion.

Reply:
Updated